### PR TITLE
[consensus] Add chain health counters

### PR DESCRIPTION
**Pull request by igor-aptos**
### Test Plan
To check counters on CI dashboard. (compat should show changing counters during restarts)

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4255)

### CODE REVIEW

**Review by zekun000 — APPROVED** [review]

File: consensus/src/counters.rs [hunk 1]
```diff
@@ -120,6 +117,65 @@ pub static LEADER_REPUTATION_ROUND_HISTORY_SIZE: Lazy<IntGauge> = Lazy::new(|| {
     .unwrap()
 });
 
+/// Computed at leader election time, with some delay.
+
+/// Window sizes for which to measure chain health.
+pub static CHAIN_HEALTH_WINDOW_SIZES: [usize; 3] = [10, 100, 1000];
```
Comment (sitalkedia) [inline consensus/src/counters.rs:123]: Let's also add 1 to the window size, which is going to give us the chain health for the last round?

File: crates/inspection-service/src/inspection_client.rs [hunk 2]
```diff
@@ -10,6 +10,31 @@ pub struct InspectionClient {
     url: Url,
 }
 
+#[derive(Clone, Debug)]
+pub enum MetricValue {
```
Comment (sitalkedia) [inline crates/inspection-service/src/inspection_client.rs:14]: Is this an unrelated change ?

File: consensus/src/counters.rs (hunk 1 above)
Comment (igor-aptos) [reply consensus/src/counters.rs:123]: 1 is pretty much useless, it is always going to be == 2f+1, not larger not smaller. 

10 looks noisy too, i.e. not all alive nodes get a chance to vote. 100 looks pretty stable.

We might want to change these to [30, 100, 300] later, but this seems fine for now.

File: crates/inspection-service/src/inspection_client.rs (hunk 2 above)
Comment (igor-aptos) [reply crates/inspection-service/src/inspection_client.rs:14]: this is required for this PR.

voting power is u128. We don't have u128 guage. So I have to use f64 guage. When I use f64 guage, this code breaks in a smoke test that checks for when voting power is u128.

Seems like I am the first one using f64 with values above u64::MAX.

**Comment by igor-aptos** [comment]
I load tested the speed of this function - it takes 2ms when computing history for 1000, for 60 nodes. not bad not great.

Also, not every round leader reputation gets called, but it gets called for enough rounds that this metrics are generally captured. Maybe instead of % 100 == 1 , I should check if last update was more than 100 rounds earlier.

**Review by sitalkedia — APPROVED** [review]
LGTM

**Comment by igor-aptos** [comment]
landing with 10, 30, 100 and 300 values. once we see how stable values look, we can tune those, if needed. From forge, I expect 100 to be the one to use for alerts and stuff.
